### PR TITLE
OCPVE-675: feat: bump api to add OLM capability

### PR DESCRIPTION
**Pull request by eggfoobar**
Bump API to include new OLM capability

Updated validation logic to make sure Marketplace can not be enabled while OLM is disabled. Marketplace has a hard dependency on OLM.

Added unit test for validation addition

/hold

Waiting on API change to be merged in. https://github.com/openshift/api/pull/1589

### CODE REVIEW

**Comment by eggfoobar** [comment]
/unhold

This should be good for a review now that API has merged in

File: pkg/types/validation/installconfig_test.go [hunk 1]
```diff
@@ -1540,6 +1540,26 @@ func TestValidateInstallConfig(t *testing.T) {
...
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Capabilities = &types.Capabilities{BaselineCapabilitySet: "None",
+					AdditionalEnabledCapabilities: []configv1.ClusterVersionCapability{"marketplace"}}
+				return c
+			}(),
+			expectedError: `additionalEnabledCapabilities: Invalid value: \[\]v1.ClusterVersionCapability{"marketplace"}: the marketplace capability requires the OperatorLifecycleManager capability`,
+		},
+		{
+			name: "invalid capability baremetal specified without MachineAPI",
```
Comment (sadasu) [inline pkg/types/validation/installconfig_test.go:1554]: Nit: This test seems unrelated to the current PR.

Comment (eggfoobar) [reply pkg/types/validation/installconfig_test.go:1554]: Apologies, I noticed we were missing one for that case, so I figured it wouldn't hurt to add one since they're a similar code path
